Sources/Components/Components/Watchlist/WKWatchlistViewModel.swift

```diff
@@ -1,27 +1,162 @@
 import Foundation
+import WKData
+import SwiftUI
 
-public final class WKWatchlistViewModel {
+public final class WKWatchlistViewModel: ObservableObject {
 
 	// MARK: - Nested Types
 
 	public struct LocalizedStrings {
 		public var title: String
 		public var filter: String
+		public var byte: String
+		public var bytes: String
 
-		public init(title: String, filter: String) {
+		public init(title: String, filter: String, byte: String, bytes: String) {
 			self.title = title
 			self.filter = filter
+			self.byte = byte
+			self.bytes = bytes
+		}
+	}
+
+	struct ItemViewModel: Identifiable {
+		public let id = UUID()
+
+		static var dateFormatter = {
+			let dateFormatter = DateFormatter()
+			dateFormatter.timeStyle = .short
+			dateFormatter.dateStyle = .none
+			return dateFormatter
+		}()
+
+		let title: String
+		let commentHTML: String
+		let commentWikitext: String
+		let timestamp: Date
+		let username: String
+		let revisionID: UInt
+		let byteChange: Int
+		let project: WKProject
+
+		public init(title: String, commentHTML: String, commentWikitext: String, timestamp: Date, username: String, revisionID: UInt, byteChange: Int, project: WKProject) {
+			self.title = title
+			self.commentHTML = commentHTML
+			self.commentWikitext = commentWikitext
+			self.timestamp = timestamp
+			self.username = username
+			self.revisionID = revisionID
+			self.byteChange = byteChange
+			self.project = project
+		}
+
+		var timestampString: String {
+			return ItemViewModel.dateFormatter.string(from: timestamp)
+		}
+
+		func bytesString(localizedStrings: LocalizedStrings) -> String {
+			// May be problematic for certain languages
```

Comment:
One approach I thought of with this is for LocalizedStrings to have a `byteChangeString: (Int) -> String` closure for the client app to define, to give it full control and avoid pluralization assumptions. Our client app would then define it as something like:

```
byteChangeString: (Int) -> String = { bytes in
String.localizedStringWithFormat(
	WMFLocalizedString("watchlist-byte-change", value:"{{PLURAL:%1$d|%1$d byte|%1$d bytes}}", comment: "Amount of bytes changed for a revision displayed in watchlist - %1$@ is replaced with the number of bytes."), bytes)
}
```

Reply:
That's a wonderful idea! I was really dissatisfied with my wrong solution, and I guess too hung up on only wanting to try to route the plurals through some heavier mechanism (like a `WKData` `WKStringUtilityService` or something). Updated here and in the client side PR. Many thanks T!